src/core/border_router/routing_manager.cpp

```diff
@@ -146,7 +146,7 @@ Error RoutingManager::LoadOrGenerateRandomOnLinkPrefix(void)
     Error error = kErrorNone;
 
     if (Get<Settings>().Read<Settings::OnLinkPrefix>(mLocalOnLinkPrefix) != kErrorNone ||
-        !IsValidOnLinkPrefix(mLocalOnLinkPrefix))
+        !IsValidOnLinkPrefix(mLocalOnLinkPrefix) || !mLocalOnLinkPrefix.IsUniqueLocal())
```

Comment:
nit: is it necessary to check `!IsValidOnLinkPrefix(mLocalOnLinkPrefix)`?
Given that local generated on-link prefix should be ULA prefix according to the specification, maybe below check is enough?

```suggestion
        !mLocalOnLinkPrefix.IsUniqueLocal())
```